### DISCUSSION TITLE
Multiple client support

### DISCUSSION THREAD

**Discussion by mattfysh** [supabase/supabase#30166] [Questions]
The `next` branch appears to be using `execProtocolRaw` instead of `execProtocol` - wouldn't this cause issues with multiple client connections? for instance, if one client attempts to perform a transaction

**Comment by GaryAustin1** [comment]
You probably need to provide more info about what repository and clients you are referring to.

**Reply by mattfysh** [reply]
oh that's odd - I was on the supabase-community/pg-gateway repo and pressed _Issues > Ask a question_ and it's brought me here, I thought I was still in the pg-gateway repo.

here is the code i'm referring to: https://github.com/supabase-community/pg-gateway/blob/next/examples/pglite-auth/password.ts#L27-L36

im a little unsure how this works with pglite, given that it's a single connection db, but pg-gateway allows multiple clients. i assume any discreet query that doesn't rely on session state or transactions works fine, but anything more than that will allow Client A to interfere with Client B, e.g. changing session parameters, or performing a query like a transaction that requires multiple messages between client and server
